io_scene_godot/converters/animation.py

```diff
@@ -0,0 +1,271 @@
+"""Export animation into Godot scene tree"""
+import copy
+import collections
+import bpy
+import mathutils
+from ..structures import NodeTemplate, NodePath, InternalResource, Array
+
+
+def fix_matrix(mtx):
+    """ Shuffles a matrix to change from y-up to z-up"""
+    # Todo: can this be replaced my a matrix multiplcation?
+    trans = mathutils.Matrix(mtx)
+    up_axis = 2
+
+    for i in range(3):
+        trans[1][i], trans[up_axis][i] = trans[up_axis][i], trans[1][i]
+    for i in range(3):
+        trans[i][1], trans[i][up_axis] = trans[i][up_axis], trans[i][1]
+
+    trans[1][3], trans[up_axis][3] = trans[up_axis][3], trans[1][3]
+
+    trans[up_axis][0] = -trans[up_axis][0]
+    trans[up_axis][1] = -trans[up_axis][1]
+    trans[0][up_axis] = -trans[0][up_axis]
+    trans[1][up_axis] = -trans[1][up_axis]
+    trans[up_axis][3] = -trans[up_axis][3]
+
+    return trans
+
+
+class KeyFrame:
+    """A keyframe contains a timestampe
+    and the conversion data"""
+    def __init__(self, timestamp, data):
+        self.timestamp = timestamp
+        self.data = data
+
+    @staticmethod
+    def identical(key_type, key1, key2):
+        """Check if two keyframes has the same
+        conversion data"""
+        if key_type == 'transform':
+            return key1.data == key2.data
```

Comment:
Is this general enough to get rid of the check for being type 'transform'?

Reply:
I keep this because I plan to do some fuzzy comparison between adjacent frames to reduce the keyframe number